src/main/java/io/debezium/connector/informix/InformixEventMetadataProvider.java

```diff
@@ -20,39 +20,29 @@ public class InformixEventMetadataProvider implements EventMetadataProvider {
 
     @Override
     public Instant getEventTimestamp(DataCollectionId source, OffsetContext offset, Object key, Struct value) {
-        if (value == null) {
+        if (value == null || source == null) {
             return null;
         }
-        final Struct sourceInfo = value.getStruct(Envelope.FieldName.SOURCE);
-        if (source == null) {
-            return null;
-        }
-        final Long timestamp = sourceInfo.getInt64(SourceInfo.TIMESTAMP_KEY);
+        final Long timestamp = value.getStruct(Envelope.FieldName.SOURCE).getInt64(SourceInfo.TIMESTAMP_KEY);
         return timestamp == null ? null : Instant.ofEpochMilli(timestamp);
     }
 
     @Override
     public Map<String, String> getEventSourcePosition(DataCollectionId source, OffsetContext offset, Object key, Struct value) {
-        if (value == null) {
-            return null;
+        if (value == null || source == null) {
+            return Map.of();
```

Comment:
Why is this different from other connectors that return `null`?

Reply:
Sorry, "magic refactoring" by IDEA, need to turn that off...